pkg/syncer/sync.go

```diff
@@ -93,30 +95,60 @@ func (s *Syncer) Sync(charts ...string) error {
 //
 // It uses topological sort to sync dependencies first.
 func (s *Syncer) SyncPendingCharts(names ...string) error {
+	var errs error
+
+	// There might be problems loading all the charts due to missing dependencies,
+	// invalid/wrong charts in the repository, etc. Therefore, let's warn about
+	// them instead of blocking the whole sync.
 	if err := s.loadCharts(names...); err != nil {
-		return errors.Trace(err)
+		klog.Warningf("There were some problems loading the information of the requested charts: %v", err)
+		errs = multierror.Append(errs, errors.Trace(err))
 	}
+	// NOTE: We are not checking `errs` in purpose. See the comment above.
+
 	charts, err := s.topologicalSortCharts()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	var errs error
-	for _, chart := range charts {
-		id := fmt.Sprintf("%s-%s", chart.Name, chart.Version)
-		// // 1. Process tgz file
-		klog.Infof("Processing %q chart...", id)
-		// // 2. Upload to target
+	if len(charts) > 1 {
+		klog.Infof("There are %d charts out of sync!", len(charts))
+	} else if len(charts) == 1 {
+		klog.Infof("There is %d chart out of sync!", len(charts))
+	} else {
+		klog.Info("There are no charts out of sync!")
+		return nil
+	}
+
+	for _, ch := range charts {
```

Comment:
I would avoid using `ch` as it may lead to confusion with Golang channels. You can use just `c` right?

Reply:
We usually use `c` for Go channels. Actually, I adopted `ch` from https://github.com/helm/helm/search?q=ch

Reply:
ok! I though we used `ch` for channels